app/views/servers/check_installation.js.erb

```diff
@@ -2,5 +2,20 @@
   $("[data-server-status~=checking]").addClass("hidden");
   $("[data-server-status~=connected]").removeClass("hidden");
 <% else %>
+  <% if @server.install_step == 1 %>
+    $(".step1").addClass("info").find("td:first-child").html("<%= j icon "circle-o-notch fa-spin" %>");
+  <% elsif @server.install_step == 2 %>
+    $(".step1").removeClass("info").addClass("success").find("td:first-child").html("<%= j icon "check-square-o" %>");
+    $(".step2").addClass("info").find("td:first-child").html("<%= j icon "circle-o-notch fa-spin" %>");
+  <% elsif @server.install_step == 3 %>
+    $(".step1").removeClass("info").addClass("success").find("td:first-child").html("<%= j icon "check-square-o" %>");
```

Comment:
@jvanbaarsen cool! btw, this reflects what I mean a bit better, but I think you get the point:

```
  $('.step').addClass("info")
  $('.step').removeClass("success")
@steps_completed.each do |step_name|
  $('.step-<%= step_name %>')removeClass("info").addClass("success")
end
```